File: app/src/main/java/eu/darken/bluemusic/bluetooth/ui/discover/DiscoverAdapter.java
package eu.darken.bluemusic.bluetooth.ui.discover;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import butterknife.BindView;
import eu.darken.bluemusic.R;
import eu.darken.bluemusic.bluetooth.core.FakeSpeakerDevice;
import eu.darken.bluemusic.bluetooth.core.SourceDevice;
import eu.darken.bluemusic.util.DeviceHelper;
import eu.darken.bluemusic.util.ui.BasicViewHolder;
import eu.darken.bluemusic.util.ui.ClickableAdapter;


class DiscoverAdapter extends ClickableAdapter<DiscoverAdapter.DeviceVH, SourceDevice> {

    @Override
    public DeviceVH onCreateBaseViewHolder(LayoutInflater inflater, ViewGroup parent, int viewType) {
        return new DeviceVH(parent);
    }

    static class DeviceVH extends BasicViewHolder<SourceDevice> {
        @BindView(R.id.name) TextView name;
        @BindView(R.id.caption) TextView caption;
        @BindView(R.id.icon) ImageView icon;

        DeviceVH(ViewGroup parent) {
            super(parent, R.layout.viewholder_device);
        }

        @Override
        public void bind(SourceDevice item) {
            super.bind(item);
            name.setText(DeviceHelper.getAliasAndName(item));
            caption.setText(item.getAddress());

            icon.setImageResource(DeviceHelper.getIconForDevice(item));

            if (item.getAddress().equals(FakeSpeakerDevice.ADDR)) {
                name.setTextColor(getColor(R.color.colorAccent));
            } else {
                name.setTextColor(getColor(android.R.color.primary_text_dark));
            }
        }
    }
}

File: app/src/main/java/eu/darken/bluemusic/bluetooth/ui/discover/DiscoverAdapter.kt
package eu.darken.bluemusic.bluetooth.ui.discover

import android.view.LayoutInflater
import android.view.ViewGroup
import eu.darken.bluemusic.R
import eu.darken.bluemusic.bluetooth.core.FakeSpeakerDevice
import eu.darken.bluemusic.bluetooth.core.SourceDevice
import eu.darken.bluemusic.bluetooth.ui.discover.DiscoverAdapter.DeviceVH
import eu.darken.bluemusic.databinding.ViewholderDeviceBinding
import eu.darken.bluemusic.util.DeviceHelper
import eu.darken.bluemusic.util.ui.BasicViewHolder
import eu.darken.bluemusic.util.ui.ClickableAdapter

class DiscoverAdapter : ClickableAdapter<DeviceVH, SourceDevice>() {
    override fun onCreateBaseViewHolder(inflater: LayoutInflater, parent: ViewGroup, viewType: Int): DeviceVH {
        return DeviceVH(parent)
    }

    class DeviceVH(parent: ViewGroup) : BasicViewHolder<SourceDevice>(parent, R.layout.viewholder_device) {

        private val ui: ViewholderDeviceBinding by lazy { ViewholderDeviceBinding.bind(itemView) }

        override fun bind(item: SourceDevice) {
            super.bind(item)
            ui.name.text = DeviceHelper.getAliasAndName(item)
            ui.caption.text = item.address
            ui.icon.setImageResource(DeviceHelper.getIconForDevice(item))
            if (item.address == FakeSpeakerDevice.ADDR) {
                ui.name.setTextColor(getColor(R.color.colorAccent))
            } else {
                ui.name.setTextColor(getColor(android.R.color.primary_text_dark))
            }
        }
    }
}